### PR TITLE
[CPU] Fixed SoftPlus for large positive values

### PR DESCRIPTION
**Pull request by a-sidorova**
### Details:
 - At this PR if SoftPlus layer gets large positive values there will be dst values (in master there is an exp overflow)
 - Added SoftRelu support from oneDNN
 - Changed spec (Added note about threshold for linear function returning)
 - Reopened [PR#4899](https://github.com/openvinotoolkit/openvino/pull/4899) (created new because branch was force-pushed)
 - PR in oneDNN: https://github.com/openvinotoolkit/oneDNN/pull/44

### Tickets:
 - 51510

### CODE REVIEW

File: inference-engine/src/mkldnn_plugin/nodes/math.cpp [hunk 1]
```diff
@@ -210,11 +210,13 @@ class MathImpl: public ExtLayerBase {
                 dst_data[i] = sinhf(src_data[i]);
             });
             break;
-        case Math::SoftPlus:
+        case Math::SoftPlus: {
+            const float large_positive_value = logf(std::numeric_limits<float>::max());
             parallel_for(dataSize, [&](size_t i) {
-                dst_data[i] = logf(expf(src_data[i]) + 1);
+                dst_data[i] = src_data[i] < large_positive_value ? logf(expf(src_data[i]) + 1.f) : src_data[i];
```
Comment (nshchego) [inline inference-engine/src/mkldnn_plugin/nodes/math.cpp:216]: Perhaps just need to divide by src_data sign?
x >= 0 ? x - ln(exp(-x) + 1) : ln(exp(x) + 1)

**Comment by a-sidorova** [comment]
@lazarevevgeny could you review spec and ngraph reference please?

File: docs/ops/activation/SoftPlus_4.md [hunk 2]
```diff
@@ -13,9 +13,23 @@
...
-SoftPlus(x) = \ln(1+e^{x})
+SoftPlus(x) = \left\{\begin{array}{r}
+    x \qquad \mbox{if } x \geq threshold \\
+    log(e^{x} + 1.0) \qquad \mbox{if } x < threshold
+\end{array}\right.
 \f]
 
+**Note**: For numerical stability the operation reverts the linear function when `x > threshold` where `threshold` depends on *T* and
+is chosen in such a way that the difference between the linear function and exact calculation is no more than `1e-6`.
+The `threshold` can be calculated with the following formula where `alpha` is the count of the numbers after the decimal point,
```
Comment (lazarevevgeny) [inline docs/ops/activation/SoftPlus_4.md:24]: `count of the numbers` -> `number of digits`

File: docs/ops/activation/SoftPlus_4.md [hunk 3]
```diff
@@ -13,9 +13,23 @@
...
 
 \f[
-SoftPlus(x) = \ln(1+e^{x})
+SoftPlus(x) = \left\{\begin{array}{r}
+    x \qquad \mbox{if } x \geq threshold \\
+    log(e^{x} + 1.0) \qquad \mbox{if } x < threshold
+\end{array}\right.
 \f]
 
+**Note**: For numerical stability the operation reverts the linear function when `x > threshold` where `threshold` depends on *T* and
```
Comment (lazarevevgeny) [inline docs/ops/activation/SoftPlus_4.md:22]: `reverts the` -> `reverts to the`

**Review by lazarevevgeny — APPROVED** [review]
Several wording comments for the spec. But in general the spec is good.

File: docs/ops/activation/SoftPlus_4.md (hunk 2 above)
Comment (a-sidorova) [reply docs/ops/activation/SoftPlus_4.md]: Thanks, done!

File: docs/ops/activation/SoftPlus_4.md (hunk 3 above)
Comment (a-sidorova) [reply docs/ops/activation/SoftPlus_4.md]: Changed, thank you!

**Review by ilyachur — APPROVED** [review]
nGraph part LGTM

**Comment by a-sidorova** [comment]
@vladimir-paramuzov @Maxim-Doronin could you review please gpu and myriad parts (disabled softplus tests)?

**Review by nshchego — APPROVED** [review]

**Review by Maxim-Doronin — APPROVED** [review]

**Comment by a-sidorova** [comment]
@dmitry-gorokhov take a look please

**Review by dmitry-gorokhov — APPROVED** [review]

**Comment by dmitry-gorokhov** [comment]
@a-sidorova please fix the conflict

**Comment by a-sidorova** [comment]
@dmitry-gorokhov done

**Pull request merged**